src/xngin/apiserver/routers/admin/test_admin.py

```diff
@@ -1447,6 +1447,37 @@ async def test_update_experiment(testing_experiment, ppatch, pget):
     assert design_spec.end_date == now + timedelta(days=1)
 
 
+@pytest.mark.parametrize("url", ["http", "http:", "http://", "http:///", "https:///", "postgres://"])
+async def test_update_experiment_url_invalid(testing_experiment, ppatch, url):
+    datasource_id = testing_experiment.datasource_id
+    experiment_id = testing_experiment.id
+    response = ppatch(f"/v1/m/datasources/{datasource_id}/experiments/{experiment_id}", json={"design_url": url})
+    assert response.status_code == 422, response.content
+    message = response.json()["detail"][0]["msg"]
+    assert message.startswith(("Input should be a valid URL", "URL scheme should be")), response.content
+
+
+@pytest.mark.parametrize(
+    ("url", "expected_url"),
+    [
+        ("http://example.com", "http://example.com/"),
+        ("https://example.com", "https://example.com/"),
+        ("https://drive.google.com/...?q=1", "https://drive.google.com/...?q=1"),
+    ],
+)
+async def test_update_experiment_url_valid(testing_experiment, ppatch, pget, url, expected_url):
```

Comment:
Can you add one more test in which a known non-empty design url is then updated with a "", resulting in the next fetch of the experiment to have `design_url is None`?